### DISCUSSION TITLE
IconLayer With base64 String?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5624] [Q&A] [answered]
I am currently using the [IconLayer](https://deck.gl/docs/api-reference/layers/icon-layer)'s `getIcon` function where I convert a base64 string into an object URL and then return the expected object structure. 
```
{
  url,
  height,
  width,
  id,
  anchorX,
  anchorY
}
```

Was curious if `getIcon` supports just returning an image via loaders.gl [parse](https://loaders.gl/modules/core/docs/api-reference/parse)/[ImageLoader](https://loaders.gl/modules/images/docs/api-reference/image-loader)?

My thinking is...

Currently, I have the base64 string version of the image and then have to convert that to an `arrayBuffer` and then to a `Blob` in order to convert it to a object url.

I would think it would be more efficient to convert the base64 string to the `arrayBuffer` and then simply return [parse](https://loaders.gl/modules/core/docs/api-reference/parse)/[ImageLoader](https://loaders.gl/modules/images/docs/api-reference/image-loader).

@Pessimistress would there be a way to do something like this with IconLayer?

**Comment by Pessimistress** [comment, marked as answer]
>  I have the base64 string version of the image and then have to convert that to an arrayBuffer and then to a Blob in order to convert it to a object url.

You should use [data URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs).

**Reply by jfuehner** [reply]
@Pessimistress I actually am using data URL currently. I believe since I am using data URL then deck is doing double-duty refetching the image. Was hoping I could maybe use the ImageLoader as I was able to with our previous [discussion](https://github.com/visgl/deck.gl/discussions/5619)?

**Reply by Pessimistress** [reply]
Auto-packing only supports loading from an URL. If you want maximum performance you should use `iconAtlas` instead.

**Reply by jfuehner** [reply]
Thanks the `iconAtlas `was exactly what I needed. Really appreciate your help.
